src/common/pmix_log.c

```diff
@@ -115,37 +111,77 @@ static void localcbfunc(pmix_status_t status, void *cbdata)
     PMIX_RELEASE(cd);
 }
 
+static void local_log(int sd, short args, void *cbdata_) {
+
+    pmix_query_caddy_t *qd = (pmix_query_caddy_t *) cbdata_;
+    pmix_info_t *data = qd->info;
+    size_t ndata = qd->ninfo;
+    pmix_info_t *directives = qd->dirs;
+    size_t ndirs = qd->ndirs;
+    pmix_info_cbfunc_t cbfunc = qd->cbfunc;
+    void *cbdata = qd->cbdata;
+    pmix_proc_t *source = qd->targets;
+    pmix_status_t rc;
+
+    /* if no recorded source was found, then we must be it */
+    if (NULL == source) {
+        pmix_proc_t *source = &pmix_globals.myid;
+        pmix_shift_caddy_t *cd;
+        cd = PMIX_NEW(pmix_shift_caddy_t);
+        cd->cbfunc.opcbfn = (pmix_op_cbfunc_t) cbfunc;
+        cd->cbdata = cbdata;
+        cd->ndirs = ndirs + 1;
+        PMIX_INFO_CREATE(cd->directives, cd->ndirs);
+        for (int n = 0; n < ndirs; n++) {
+            PMIX_INFO_XFER(&cd->directives[n], (pmix_info_t *) &directives[n]);
+        }
+        PMIX_INFO_LOAD(&cd->directives[ndirs], PMIX_LOG_SOURCE, source, PMIX_PROC);
+        /* call down to process the request - the various components
+         * will thread shift as required */
```

Comment:
We already did the thread shift - probably should delete this from the comment to avoid future confusion